chutney/ui/src/app/core/guards/auth.guard.ts

```diff
@@ -12,18 +12,26 @@ import { TranslateService } from '@ngx-translate/core';
 import { LoginService } from '@core/services';
 import { AlertService } from '@shared';
 import { Authorization } from '@model';
+import { firstValueFrom } from "rxjs";
 
 
-export const authGuard: CanActivateFn = (route: ActivatedRouteSnapshot, state: RouterStateSnapshot) => {
+export const authGuard: CanActivateFn = async (route: ActivatedRouteSnapshot, state: RouterStateSnapshot) => {
```

Comment:
I need the async keyword because i need to await for the result of the ws that fetches the current user to authenticate via SSO and then to continue the workflow of authentication. I can refacto the guard with observable like this ?

return of(loginService.isAuthenticated()).pipe(
    switchMap((isAuthenticated: boolean) => {
      if (isAuthenticated) {
        return of(true);
      } else if (loginService.oauth2Token) {
        return loginService.initLoginObservable(requestURL, {
          'Authorization': 'Bearer ' + loginService.oauth2Token
        }).pipe(
          switchMap(() => of(true)),
          catchError(() => {
            loginService.initLogin(requestURL);
            return of(false);
          })
        );
      } else {
        loginService.initLogin(requestURL);
        return of(false);
      }
    }),
    switchMap((isLoggedIn: boolean) => {
      if (!isLoggedIn) {
        return of(false);
      }

      const authorizations: Array<Authorization> = route.data['authorizations'] || [];
      if (loginService.hasAuthorization(authorizations)) {
        return of(true);
      } else {
        alertService.error(unauthorizedMessage, { timeOut: 0, extendedTimeOut: 0, closeButton: true });
        loginService.navigateAfterLogin();
        return of(false);
      }
    })
  );
};

Reply:
No keep async for the moment